backend/migrations/README.md

```diff
@@ -0,0 +1,111 @@
+# Managing DB with Alembic locally
+In data.all we use [Alembic](https://alembic.sqlalchemy.org/en/latest/)  -- a lightweight database migration tool for usage with the SQLAlchemy Database Toolkit for Python.
+Alembic relies on the database's current state to generate and apply migrations accurately.
+Alembic determines the changes to be made by comparing the current state of the database with the desired state specified in your SQLAlchemy models. This process, known as schema diffing, requires an existing database to identify differences.
+Alembic generates migration scripts based on the detected differences between the current database schema and the desired schema defined in your application code. This generation is dependent on the actual structure and content of the database.
+
+In order to create and test migrations locally you will have to create a local database.
+
+## Prerequisites
+
+1. Build and launch Docker containers for the database and GraphQL. 
+```bash
+docker compose build db
+docker compose run db
+docker compose build graphql
+docker compose run graphql
+```
+These can also be initiated alongside all local testing containers using the following command:
+```bash
+docker compose up
+```
+2. Specify the location of database model descriptions. If you are at the project's root, the target folder path is backend.
+```bash
+export PYTHONPATH=backend
+```
+3. The containers initiated in the first step will default to using the schema named `local`. Alembic relies on the environmental variable `envname` to determine the schema. Set it to `local` with the following command:
+```bash
+export envname=local
```

Comment:
I am afraid that the containers will use the schema named `dkrcompose` and not `local` as per the docker-compose.yaml file

Reply:
Yes, if using the docker containers the `envname` is `dkrcompose` because it is specified in the graphql container:
<img width="577" alt="image" src="https://github.com/data-dot-all/dataall/assets/71252798/359b85f2-dea7-4ce9-b0de-6c38a3ae1a3f">

Reply:
Agree with some of the earlier comments - I think we should to explicitly call out the different schemas created (`dkrcompose` to be used in local deploy of data.all vs. `local` for isolated instance of db schema to play with)

- Running data.all following local deploy commands will have db schema built using declarative_base (I believe from the line - `Base.metadata.create_all(engine.engine)` 

- Running `alembic -c backend/alembic.ini upgrade head` with a new envname (such as `local`) to build schema from scratch

Let's maybe separate the 2 above and highlight the use case for working with each?

Also, a call out but when I run an upgrade command `alembic -c backend/alembic.ini upgrade head` with envname `dkrcompose` to make changes to local deploy dataall db changes - I need to also edit the hostname to `localhost` because of this line in `get_engine()` - `hostname = 'db' if envname == 'dkrcompose' else 'localhost'` (and when I change hostname then other docker containers can not connect until I revert back to `db`)

...not sure if others do the same / something we could make easier as part of this PR

Reply:
Hi @noah-paige , Agree on the point that we need to explicitly mentioned which schema does what. On your last point, I have faced that issue while upgrading the dockercompose db and use the same trick as you do. It would be very helpful if we solve this in this PR